osu.Game.Rulesets.Catch/Edit/CatchHitObjectUtils.cs

```diff
@@ -20,5 +23,41 @@ public static Vector2 GetStartPosition(ScrollingHitObjectContainer hitObjectCont
         {
             return new Vector2(hitObject.OriginalX, hitObjectContainer.PositionAtTime(hitObject.StartTime));
         }
+
+        /// <summary>
+        /// Get the range of horizontal position occupied by the hit object.
+        /// </summary>
+        /// <remarks>
+        /// <see cref="TinyDroplet"/>s are excluded and returns <see cref="PositionRange.EMPTY"/>.
+        /// </remarks>
+        public static PositionRange GetPositionRange(HitObject hitObject)
+        {
+            switch (hitObject)
+            {
+                case Fruit fruit:
+                    return new PositionRange(fruit.OriginalX);
+
+                case Droplet droplet:
+                    return droplet is TinyDroplet ? PositionRange.EMPTY : new PositionRange(droplet.OriginalX);
+
+                case JuiceStream _:
+                    return GetPositionRange(hitObject.NestedHitObjects);
+
+                case BananaShower _:
+                    // A banana shower occupies the whole screen width.
+                    return new PositionRange(0, CatchPlayfield.WIDTH);
```

Comment:
This function is used in other places apart from flip where this makes sense, but it also has the side effect of including the banana shower's width in a flip operation. I'm not sure if this is expected. In the osu! editor for example, we exclude spinners from flip operations currently. Although maybe that's wrong, since we are eventually going to unlock spinner positioning (and potentially the same for banana showers, limiting them to a subregion of the available width?).

Probably fine to go with this as is - just pointing it out.